velox/functions/prestosql/ArrayContains.cpp

```diff
@@ -104,10 +104,10 @@ void applyComplexType(
 
     auto searchIndex = searchIndices[row];
     for (auto i = 0; i < size; i++) {
-      if (elementsBase->isNullAt(offset + i)) {
+      if (elementsDecoded.isNullAt(offset + i)) {
         foundNull = true;
       } else if (elementsBase->equalValueAt(
```

Comment:
same comments as above not sure why did we decode the elements.

Reply:
The elements may be dictionary/constant encoded, for example, an array vector with elements from a dictionary-wrapped map vector in the test case.

```C++
// ArrayVector with DictionaryVector<Array> elements.
  auto baseVector =
      makeArrayVector<int64_t>({{1, 2, 3, 4}, {3, 4, 5}, {10, 9, 8, 7}});
  auto baseVectorSize = baseVector->size();
  const vector_size_t kTopLevelVectorSize = baseVectorSize * 2;
  BufferPtr indices =
      AlignedBuffer::allocate<vector_size_t>(kTopLevelVectorSize, pool_.get());
  auto rawIndices = indices->asMutable<vector_size_t>();
  for (size_t i = 0; i < kTopLevelVectorSize; ++i) {
    rawIndices[i] = i % baseVectorSize;
  }
  auto dictVector = BaseVector::wrapInDictionary(
      nullptr, indices, kTopLevelVectorSize, baseVector);
  auto arrayVector = makeArrayVector({0, baseVectorSize + 1}, dictVector);
```

Reply:
either would have worked as explained in the comment above, however i do not have string opinion so you are free to select what you feel more suitable